### DISCUSSION TITLE
Dynamic Graphs - Looking for dynamic .gexf graph files, plus an inquiry about JS tool support.

### DISCUSSION THREAD

**Discussion by Nikovic** [gephi/gexf#22] [General]
Greetings!
I'm in the process of writing my bachelor's thesis on dynamic network visualization (using adjacency matrices), and found the .gexf format to be the most elegant (and only?) file format for representing dynamic graphs. Kudos to everyone involved!
(edit: I remembered that I've also found [.dgs](https://graphstream-project.org/doc/Advanced-Concepts/The-DGS-File-Format/) for representing dynamic graphs. See [this thread](https://github.com/gephi/gexf/issues/14#issuecomment-1722186772) for more)

Now I have two questions: Does anyone here have any dynamic networks in .gexf format "lying around"? I've seen the small example in the v1.3 primer, but would be eager to get my hands on some more elaborate, ideally real-world datasets.

And secondly: I've seen the JavaScript tools listed on the gexf.net website, but from a cursory glance, none of them deal with dynamic networks. Is that impression correct?

**Comment by Nikovic** [comment]
I'll partially answer my own question: On https://github.com/gephi/gephi/wiki/Datasets, one can find one dynamic network ([download link](https://gephi.org/datasets/ht2009_15min.gexf.gz)). It's called "Hypertext 2009 dynamic contact network: contact network during the Hypertext 2009 conference". It's slightly misshaped (has two meta tags), but it's real data one can play with :)

**Reply by mbastian** [reply]
Hi @Nikovic, 

For JS, yes the reference library is [graphology](https://www.npmjs.com/package/graphology-gexf) and it doesn't support dynamic graphs. It would probably be a significant effort to add this, but @Yomguithereal can comment further, as he is the maintainer.

For datasets, here is some toy examples I used for testing, hopefully that helps
[gexf1.3.zip](https://github.com/gephi/gexf/files/13624301/gexf1.3.zip)

For real datasets, I would need to do more digging, but here is one from sociopatters: 
[ht2009_15min.gexf.zip](https://github.com/gephi/gexf/files/13624341/ht2009_15min.gexf.zip)

**Reply by Yomguithereal** [reply]
Yes, I don't really support dynamic graphs. The main reason being there is no single generic way to represent those in memory, especially if you want said representation to be efficient. This said I am not opposed to adding some way to output the dynamic graph information as raw JSON at one point, but as you said, this would represent a significant effort that has no direct benefit to me or my research community as of yet (which does not mean we are not interested in dynamic graphs, just that we work with those on a case to case basis).
